src/AllocationBoundsInference.cpp

```diff
@@ -86,6 +86,17 @@ class AllocationInference : public IRMutator {
                 max = b[i].max;
                 extent = simplify((max - min) + 1);
             }
+            if (bound.modulus.defined()) {
+                internal_assert(bound.remainder.defined());
+                min -= bound.remainder;
+                min = (min / bound.modulus) * bound.modulus;
+                min += bound.remainder;
+                Expr max_plus_one = max + 1;
+                max_plus_one -= bound.remainder;
+                max_plus_one = ((max_plus_one + bound.modulus - 1) / bound.modulus) * bound.modulus;
+                max_plus_one += bound.remainder;
+                extent = simplify(max_plus_one - min);
```

Comment:
Would it work to say the extent is just ((max - min + 1) / bound.modulus) * bound.modulus, where the min has already been aligned? It seems like this would produce the minimal extent required to cover the max, and would generate simpler code?